src/App.jsx

```diff
@@ -14,45 +19,140 @@ export const goodsFromServer = [
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    reversedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good2.localeCompare(good1);
+
+        case SORT_FIELD_LENGTH:
+          return good2.length - good1.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return reversedGoods;
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods
+  = getPreparedGoods(goodsFromServer, { sortField, isReversed });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        {sortField === SORT_FIELD_NAME
+          ? (
+            <button
+              type="button"
+              id="name"
+              className="button is-info"
+              onClick={() => setSortField(SORT_FIELD_NAME)}
+            >
+              Sort alphabetically
+            </button>
+          )
+          : (
+            <button
+              type="button"
+              id="name"
+              className="button is-info is-light"
+              onClick={() => setSortField(SORT_FIELD_NAME)}
+            >
+              Sort alphabetically
+            </button>
+          )}
+
+        {sortField === SORT_FIELD_LENGTH
+          ? (
+            <button
+              type="button"
+              id="name"
+              className="button is-success"
+              onClick={() => setSortField(SORT_FIELD_LENGTH)}
+            >
+              Sort by length
+            </button>
+          )
+          : (
+            <button
+              type="button"
+              id="name"
+              className="button is-success is-light"
+              onClick={() => setSortField(SORT_FIELD_LENGTH)}
+            >
+              Sort by length
+            </button>
+          )}
+
+        {isReversed
+          ? (
+            <button
+              type="button"
+              id="reverse"
+              className="button is-warning"
+              onClick={() => setIsReversed(!isReversed)}
+            >
+              Reverse
+            </button>
+          )
+          : (
+            <button
+              type="button"
+              id="reverse"
+              className="button is-warning is-light"
+              onClick={() => setIsReversed(!isReversed)}
+            >
+              Reverse
+            </button>
+          )}
```

Comment:
```suggestion
            <button
              type="button"
              id="reverse"
              className={cn("button is-warning", { isLight: !isReversed })}
              onClick={() => setIsReversed(prevIsReversed => !prevIsReversed)}
            >
              Reverse
            </button>
```
 use classnames library for dynamic set classNames. no need repeat jsx for it.
 same for other buttons
 
better change isReversed from prev state